core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -744,4 +790,40 @@ public boolean accept(File userStores, String name) {
         }
 
     }
+
+    /**
+     * Check for tenant domain contains any trailing spaces.
+     *
+     * @param tenantDomain
+     */
+    private void tenantDomainNameValidation(String tenantDomain) {
+
+        if (tenantDomain.equals(tenantDomain.trim())) {
+            if (log.isDebugEnabled()) {
+                log.debug("Tenant domain doesn't contain any trailing white spaces, tenantDomain: {" + tenantDomain
+                        + "}");
+            }
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Tenant domain contains trailing white spaces, tenantDomain: {" + tenantDomain + "}, "
+                        + "current stack trace: \n" + printCurrentStackTrace().toString());
+
+            }
+        }
+    }
+
+    /**
+     * Print current stack trace.
+     */
+    private StringBuilder printCurrentStackTrace() {
+
+        StackTraceElement[] elements = Thread.currentThread().getStackTrace();
+        StringBuilder currentStackTraceBuilder = new StringBuilder();
+        for (int i = 1; i < elements.length; i++) {
```

Comment:
Can the elements null? pls check

Reply:
This getStackTrace() method will return a zero-length array if this thread has not started, has started but has not yet been scheduled to run by the system, or has terminated.
hence it won't be null.
[1] https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#getStackTrace()